tools/docker-entrypoint.sh

```diff
@@ -2,18 +2,32 @@
 
 : "${EXPOSE_TCP:=false}"
 
+# SIGTERM-handler this function gets called on Ctrl+C or if docker trys to stop the container
+term_handler(){
+   echo "*** Stopping C-Lightning Node ***"
+   lightning-cli stop # this will gracefully shutdown the node and then exit
+   exit 0
+}
+
+# Setup signal handlers
+trap 'term_handler' SIGTERM SIGINT
+
 if [ "$EXPOSE_TCP" == "true" ]; then
     set -m
     lightningd "$@" &
-
+    pid="$!"
     echo "C-Lightning starting"
     while read -r i; do if [ "$i" = "lightning-rpc" ]; then break; fi; done \
     < <(inotifywait  -e create,open --format '%f' --quiet "$LIGHTNINGD_DATA" --monitor)
     echo "C-Lightning started"
     echo "C-Lightning started, RPC available on port $LIGHTNINGD_RPC_PORT"
 
     socat "TCP4-listen:$LIGHTNINGD_RPC_PORT,fork,reuseaddr" "UNIX-CONNECT:$LIGHTNINGD_DATA/lightning-rpc" &
-    fg %-
 else
-    exec lightningd "$@"
+    exec lightningd "$@" &
```

Comment:
this make no sense, because anything executed after the `exec` will never be run.